code/modules/research/weaponsanalyzer.dm

```diff
@@ -0,0 +1,282 @@
+/obj/machinery/weapons_analyzer
+	name = "Weapons Analyzer"
+	icon = 'icons/obj/machines/research.dmi'
+	icon_state = "weapon_analyzer"
+	density = 1
+	anchored = 1
+	use_power = 1
+	idle_power_usage = 60
+	active_power_usage = 2000
+
+	var/obj/item/gun/gun = null
+	var/obj/item/item = null
+	var/obj/item/device/laser_assembly/assembly = null
```

Comment:
Would it not be better to just have one variable cast as obj then typecast further when required?